src/main/java/org/mule/extensions/jms/api/source/ProcessingStrategy.java

```diff
@@ -0,0 +1,52 @@
+package org.mule.extensions.jms.api.source;
+
+import org.mule.extensions.jms.api.config.ConsumerAckMode;
+
+import javax.jms.Message;
+import javax.jms.Session;
+
+/**
+ * Determines how the {@link Message messages} should be consumed by a {@link Message} listener.
+ *
+ * @since 1.0
+ * @see Session
+ * @see Message#acknowledge()
+ */
+public enum ProcessingStrategy {
+
+  /**
+   * Makes the listener work synchronously. Only one {@link Message} at a time will be consumed, delivered
+   * and waited to be processed in the flow.
+   * At any time, only one {@link Message} will be in the processing pipeline for any given {@link Session}.
+   * <p>
+   * If using the {@link ConsumerAckMode#AUTO} ACK mode, this processing strategy guarantees the ACK
+   * to be performed on each {@link Message} exclusively when the flow execution is completed.
+   */
+  SYNCHRONOUS(true),
+
+  /**
+   * Makes the listener work asynchronously. As soon as one {@link Message} is consumed and delivered,
+   * the listener is freed to consume the next available {@link Message}, thus multiple {@link Message messages}
+   * can be in the processing pipeline at the same time for a given {@link Session}.
+   * <p>
+   * If using the {@link ConsumerAckMode#AUTO} ACK mode, this processing strategy will perform the ACK
+   * on each {@link Message} as soon as it is delivered to the flow, regardless of whether its
+   * execution is successful or not.
+   * <br>
+   * Since multiple {@link Message messages} can be in flight for the same {@link Session}, whenever an ACK
+   * is performed on any of those {@link Message messages}, <b>all the in-flight messages will be ACKd</b>,
+   * regardless of the processing result of each of them.
+   */
+  ASYNCHRONOUS(false);
```

Comment:
I don't see an use case for ASYNCHRONOUS. Or I would like to process messages in order, or I would like to process as many messages as possible concurrently. And Auto ACK mode should behave the same regardeless of the strategy.